[Feature] Support Form Iteration

kimberleyworld opened the pull request:
Created a new version on the donation form, separated it into different render functions to accommodate the different layouts based on screen size. 

Created a version for the new pages which takes up only half of the page. 

Added new meta boxes to allow for dynamic regular and one-off text to override the standard input boxes.

kimberleyworld committed:
Built main outline of new support box

kimberleyworld committed:
Added payment options in

kimberleyworld committed:
Update lib version

kimberleyworld committed:
New hover state

kimberleyworld committed:
Merge branch 'development' into feature-update-donate-form

kimberleyworld committed:
xxl screen size format

kimberleyworld committed:
Remove border around input

kimberleyworld committed:
render function to output payment options

kimberleyworld committed:
Funtional version of form

I've tried to create multiple functions to make up the donation form so that I can make it responsive. The layout changes so much between screens that this seems the only way to do it. This is the bare bones,  need to edit further.

kimberleyworld committed:
Started to style mobile form

added in breakpoint variable, styles the mobile version of the form

kimberleyworld committed:
remove old margin

found old margin class

kimberleyworld committed:
payment positioning and mobile schedule tabs

kimberleyworld committed:
Fixed issues with reponsivness

kimberleyworld committed:
Removed flex end for left column desktop

kimberleyworld committed:
initial commit of condensed version

kimberleyworld committed:
fixed the schedule active buttons

Attempted to fix carry over issue, needs another look

kimberleyworld committed:
fixed increased padding on desktop

kimberleyworld committed:
fixed double setting of url param and started to format options-fundraising page

kimberleyworld committed:
Formatting the fundraising options page

kimberleyworld committed:
one-off and regular admin text boxes rendering but not changing as schedule button it changed

kimberleyworld committed:
Updates heading and text once but not on following button clicks

kimberleyworld committed:
Heading and description updating on click of oneoff and regular

kimberleyworld committed:
Removed console and redundant value variable

kimberleyworld committed:
Created a dispatcher function to keep code DRY

kimberleyworld committed:
Remove small text below buttons

kimberleyworld committed:
Fixed hover issue on tabs and resizing issue for condensed version

kimberleyworld committed:
Trying to fix dynamic headings

Swopped them from id's
now works when condensed and mobile present but not condensed and desktop

kimberleyworld committed:
Fixed dynamic headings and text

Deleted the .first in the js so that the desktop and mobile headings are changed.

kimberleyworld committed:
Moved images to src.

Moved images to src.

kimberleyworld committed:
Removed condensed version for design review

kimberleyworld committed:
Updated the labels on the admin text boxes to make it clearer

kimberleyworld committed:
Updated Payment images

kimberleyworld committed:
Changed src to dist and corrected paypal typo

kimberleyworld committed:
removed comment

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 91b8bdc according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

varffs commented on dist/img/support-form/Mastercard.png:
These icon files seems low resolution not just small. They render a bit fuzzy on screen too. I think we need higher quality version probably at 2x the output size

varffs replied:
SVG file would probably be preferable here too tbh

kimberleyworld replied:
Can't export svg's at 2x output but have swopped them to svg and looks way better

varffs commented on lib/renderers.php:
```diff
@@ -25,61 +25,236 @@ function ( $v ) {
+  <div class="<?php echo esc_attr( $button_classes ); ?>">
+    <div class="grid-row grid--nested-tight mb-5">
+      <!-- Low, Medium, High Tier Buttons -->
+      <?php foreach ( array( 'low', 'medium', 'high' ) as $tier ) : ?>
```
I wonder if this is a php lint config thing (either wrong rule or lacking a rule) but we dont do the colon style php conditionals or loops anywhere in the codebase. So these need to all be the curly bracket style { open and closed } loops

varffs commented on lib/renderers.php:
```diff
@@ -25,61 +25,236 @@ function ( $v ) {
+
+  if ( ! empty( $heading_text_data ) ) {
+    echo '<script>';
+    echo 'window.SupportFormCopy = ' . wp_json_encode( $heading_text_data ) . ';';
```
we dont need to do this here and attach to the window global (this will also be rendered into that global multiple times if there are multiple forms on the page). We can use the approach used for the autovalues data that is set on a globally available javascript object we already have called `WP`. 

You can find the code that does this action inside functions.php L11. On that line we create an object available to our javascript that we use when we need to pass data like you have done here. You will notice the usage of `nm_get_support_autovalues` there, but you can just extend the generation of that object with your data from `$heading_text_data`

kimberleyworld replied:
Understood. Thanks, have noted it in a data architecture guidelines doc for myself

varffs commented on lib/renderers.php:
```diff
@@ -25,61 +25,236 @@ function ( $v ) {
+    </a>
+    <?php if ( $text ) : ?>
+      <div class="mb-5">
+        <a href="<?php echo esc_url( home_url( 'support/' ) ); ?>" class="js-fix-widows support-form__dynamic-text">
```
We almost certainly dont want fix widows utility here, unless specifically spec'd by design

varffs commented on lib/renderers.php:
```diff
@@ -25,61 +25,236 @@ function ( $v ) {
+  $img_base = get_template_directory_uri() . '/dist/img/support-form/';
+  ?>
+  <div class=" <?php echo esc_attr( $payment_classes ); ?>">
+    <img class="support-form__payment-type mr-2" src="<?php echo $img_base; ?>Visa.png" alt="visa icon" />
```
may or may not be worth it but this kind of repeated very similar content is ripe for replacement with an array and a loop

kimberleyworld replied:
swopped for array and loop, makes sense

varffs commented on lib/renderers.php:
```diff
@@ -25,61 +25,236 @@ function ( $v ) {
+      <div class="p-5">
+        <?php render_support_heading_and_text( $donation_mode, 'is-s-24' ); ?>
+            <?php render_support_form_amount_buttons( $active_values, $instance, '' ); ?>
+        <?php render_payment_icons( 'mt-3 support-form__payment-type-condensed' ); ?>
```
if the only parameters are the classes for wrapping the icons why not just include the wrapping div in here and have a parameter-less function call inside that div? I think it might make the code more readable at this level.

kimberleyworld replied:
Ah that's intentional. The wrapper div and its classes are needed to control the layout in different contexts (mobile, desktop, condensed). Moving the wrapper outside and making the function parameter-less would lock the layout and reduce flexibility. Passing wrapper classes keeps the icon markup DRY and allows easy styling changes based on where the icons are used. Let me know if you can see a way to do that with the wrapper outside, I can't!

varffs replied:
I meant something like:

```
<div class='mt-3 support-form__payment-type-condensed'>
  <?php render_payment_icons(); ?>
</div>
```

to me I think I found that a bit easier to read, but i dont feel strongly about it

varffs commented on lib/renderers.php:
```diff
@@ -25,61 +25,236 @@ function ( $v ) {
+  $instance = uniqid( 'support-form-' );
+  $support_section_autovalues = nm_get_support_autovalues();
+  $active_values = $support_section_autovalues['default'];
+  $donation_mode = nm_get_donation_mode( $active_values );
```
as this function `nm_get_donation_mode` is only called once here in the dispatcher i think it probably makes more sense to keep this logic inside that dispatch function. (tho perhaps with the window global to WP global refactor you might not even need the logic anymore)

varffs commented on lib/theme-options/options-fundraising.php:
```diff
@@ -9,262 +9,366 @@ function nm_register_fundraising_options_metabox() {
+
+        $main_options->add_field(
+            array(
+                'name' => 'scheduled Donation Overrides',
```
I think maybe `scheduled` is a confusing word here. I thought it might have meant something that was temporarily shown during fundraisers or something related to time conditionals. `Override` is probably also essentially technical. Just something like `Support section dynamic copy` would work. The name generally used for this component is `Support section`

varffs commented on partials/support-section.php:
```diff
@@ -1,43 +1,5 @@
-  </div>
-</div>
\ No newline at end of file
+<div>
```
This needs a `container` class as because the element is now boxed with the background not full width its lost it

kimberleyworld replied:
I intentaiallly removed the container so it would be full width as in the new designs, but in the latest design feedback there was talk about how to make it more cohesive. This could be a shout. I'll implement and see what they think.

varffs commented on src/js/modules/Support.js:
```diff
@@ -68,6 +68,18 @@ export class Support {
 
           const $button = $(this);
           const data = $button.data();
+          // function to update the support section copy depending on the type of donation
```
My instinct here would have been toggling an overall modifier class to change the visibility of already existing elements. I would expect that to be (obviously extremely fractionally) more performant and I think it is a standard practice. If it doesn't matter or we do this jquery inserting often across the codebase then this approach is ok.

kimberleyworld replied:
So my understanding is that because it's dynamic text that changes semi-regularly, swapping the inner HTML is the most straightforward and that preloading all possible variations in the markup (and toggling visibility) won't work here because we don’t know the content ahead of time? I think the performance hit is pretty negligible

varffs replied:
The content should be the same source as that which renders the initial markup. But im fine with this solution. However we probably should use pure javascript or the jquery `.text()` function rather than the full html insert as that is def a small performance hit.

kimberleyworld replied:
perf, added .text instead

varffs commented on src/styl/layouts/support-form.styl:
```diff
@@ -9,22 +9,110 @@
-    background-color: var(--color-red)
-    border-color: var(--color-white-soft)
+.support-form__box-radius
+  border-radius: 4px
```
this shouldn't need its own class as we have radius variables in the style library. So using the library class (or css variable if needed) will keep consistency across the site. If design have used an inconsistent sizing thats their mistake (but should be flagged to them)

kimberleyworld replied:
I've added this to[ the design review doc](https://www.notion.so/novaramedia/Support-module-design-review-1eda422f298f805caef6ea860da7a965), to see what we wanna do. The radius variable in the lib is way smaller. I created a   `--corner-radius-large: 4px variable` and `ui-rounded-box-large ` class but forgot to implement it. Either way some design decision is needed.

varffs commented on src/styl/layouts/support-form.styl:
```diff
@@ -9,22 +9,110 @@
+  text-align: center;
+  transition: opacity var(--transition-speed-ultra);
+  &:not(&--active):hover:not(:focus)
+    opacity: 0.5;
```
This doesnt match the button hover styles and should. These inputs should behave just like any other buttons across the site. Hopefully there is a way that you can use aspects of the library in order to have this output. 

In general it would be good to make a little of the form input styles unique but have everything come from library defaults. But obviously for the joined form element that won't be immediately possible

kimberleyworld replied:
yeh there was a comment about this in the design review too. i was finding it tricky to get it to look right from the library but will have another go

kimberleyworld replied:
i need to build some new class variants into the UI lib

varffs commented on src/styl/layouts/support-form.styl:
```diff
@@ -9,22 +9,110 @@
+.support-form__buttons-mobile
+  display: none;
+
+.support-form__left-column-desktop, .support-form__right-column-desktop
```
You should be able to avoid this by using the nested modifiers for the grid classes. I think this is the correct class to reference: https://github.com/novaramedia/nm-stylus-library/blob/main/functions/grid-function-new.styl#L42 (However it also looks to me that that class is badly named in the library and should really be `grid-row--nested`, but I'm not 100% sure rn)

kimberleyworld replied:
i added this to get it closer to the new design, the nested padding is too large

varffs replied:
If that is the case then either the nesting should have options added to it in the library, or possibly the design is too far from the existing system. I think it matters because the nesting stuff is all to ensure correct vertical alignment with the grid system in whatever usage

varffs replied:
Looking at this on my local It seems to be making the middle of the support form the exact middle of the screen, which means it doesnt align with the gutter in the grid system

varffs commented on src/styl/layouts/support-form.styl:
```diff
@@ -9,22 +9,110 @@
+
+.support-form__payment-type
+  vertical-align: bottom
+  height: 3vh;
```
This should probably have a relative measure related to type sizes rather than window height, even if this means some additional responsive classes. Otherwise the display is fairly unpredictable

varffs reviewed (changes requested):
Very nearly there I think. Just a few details of aspects kinda deep into the codebase (that I think will make the final code tighter).

Not sure if you agree from working on it but I do think that the design on mobile of the tab is a bit off. It doesn't really feel either obvious or easy to understand how to switch between the donation types, or even really that it is possible. (Understand this is out of your scope building the thing to choose). Obv I'm not going to block the PR due to things like that but thought I'd flag here as an additional pair of eyes

kimberleyworld commented:
Great, thanks for the feedback. I actually like the tabs. I think it looks quite sleek and trustworthy. There was talk about what to do about the white on white background elsewhere in the site (on new pages the whole page is grey so it stands out more and looks better imo) so I will chase that. 

I have been collecting design review feedback i[n this document](https://www.notion.so/novaramedia/Support-module-design-review-1eda422f298f805caef6ea860da7a965?source=copy_link). I have copied across your design related feedback so that it can be discussed. i.e. tabs and radius.

kimberleyworld committed:
Added payment types as svgs

kimberleyworld committed:
extend existing global objects instead of window global for supportSectionCopy

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 061e0f0 according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

kimberleyworld committed:
some functional adjustments from code review

 - removed js-fix-windows from form
- swopped payment icons into a loop
- moved nm_get_donation_mode() into the dispatcher function
-

kimberleyworld committed:
updating button ui

kimberleyworld committed:
changed payment icons to rem

kimberleyworld committed:
trying to get the right hover effect on tabs

kimberleyworld committed:
Added padding to the text

kimberleyworld committed:
Hover state of active classes

kimberleyworld committed:
fixed hover while active issue

kimberleyworld committed:
custom input box wider

kimberleyworld committed:
Prevented enter key submitting custom-input

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 952e467 according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

varffs commented on lib/renderers.php:
```diff
@@ -25,61 +25,236 @@ function ( $v ) {
+  <div class="<?php echo esc_attr( $button_classes ); ?>">
+    <div class="grid-row grid--nested-tight mb-5">
+
+      <!-- Low, Medium, High Tier Buttons -->
```
I think before production would be good to remove code comments that render. maybe I'm wrong? maybe we can find a script way to do this? alternatively you can use php blocks for comments in PHP that mean they won't render anything front end. or are they useful in the HTML when debugging?

varffs commented on lib/renderers.php:
```diff
@@ -25,61 +25,236 @@ function ( $v ) {
+      <h4 class="support-form__dynamic-heading font-size-13 font-weight-bold mb-3">
+        <?php echo esc_html( $heading ); ?>
+      </h4>
+    <?php if ( $text ) : ?>
```
another colon not curly conditional to catch

varffs commented on lib/renderers.php:
```diff
@@ -25,61 +25,236 @@ function ( $v ) {
+  );
+  ?>
+  <div class="<?php echo esc_attr( $payment_classes ); ?>">
+    <?php foreach ( $payment_methods as $filename => $alt_text ) : ?>
```
colon conditional

varffs commented on partials/support-section.php:
```diff
@@ -1,43 +1,6 @@
-</div>
\ No newline at end of file
+<div class="container">
+     <?php
```
I know this is just for testing, but in the final output the support section should always be used inside a grid and row layout

varffs commented on src/js/modules/Support.js:
```diff
@@ -68,6 +68,18 @@ export class Support {
+            const heading = $form.find('.support-form__dynamic-heading');
+            const text = $form.find('.support-form__dynamic-text');
+
+            if (heading.length) heading.html(copy.heading);
```
I think full syntax here with brackets is easier to read and a better option even if this works completely fine. 

However there also needs to be a fallback solution here in case the metadata isn't set like in the PHP function. When texting locally (and possibly without the updated metadata saved) I end up with forms displaying no text when I toggle. So even if this is to catch unlikely data moments I think better to fallback to the defaults when no override text is set.

varffs commented on src/js/modules/Support.js:
```diff
@@ -77,18 +89,20 @@ export class Support {
 
             $('.support-form__custom-input').removeClass(
-              'support-form__button--active'
+              'support-form__button--active ui-button--active'
```
is the `support-form__button--active` class doing anything now? Probably if we have a ui-button active class we can avoid having both of them and just target that class as the active state

kimberleyworld replied:
I know it's there for a reason, but I can't remember what that is, I need to dig into what the reason is and if I can change it

varffs commented on src/js/modules/Support.js:
```diff
@@ -101,11 +115,15 @@ export class Support {
+        },
+        keydown(event) {
+          if (event.key === 'Enter') {
+            event.preventDefault(); // prevents form submission on enter and state reset
```
This is probably fine, but we should check for accessibility / screen readers on this as might prevent some people from being able to donate. Actually a good point that we should do a pass over the entire form for accessibility stuff. Probably means a load of aria labelling at least.

kimberleyworld replied:
I'm going to do an a11y checklist and document here : https://www.notion.so/novaramedia/Support-module-design-review-1eda422f298f805caef6ea860da7a965?source=copy_link#210a422f298f807e8a29eb84d3c5c975

kimberleyworld replied:
Adding in arial labels

varffs commented on src/styl/layouts/support-form.styl:
```diff
@@ -9,22 +9,109 @@
+
+.support-form__button
+  &--active
+    color: var(--color-black-soft);
```
is this fixing or overriding ui-button for this context?

kimberleyworld replied:
fixing, there is not yet an active button state and I didn't want to create one just for this varrient

varffs commented on src/styl/layouts/support-form.styl:
```diff
@@ -9,22 +9,109 @@
+  justify-content: space-between;
+  flex-wrap: wrap;
+
+.support-form-condensed
```
I think this isn't how the condensed should work, as it wont then be able to fix into any grid layout we try. Unless this is just testing for build. I notice that the condensed form seems to be the same as the mobile view? If that is the case I think there is a viable refactor to basically render the mobile version at all screen sizes as the condensed version

kimberleyworld replied:
There is defo a reason I did it this way, i will dig into why that was

kimberleyworld replied:
So i've worked out that I was trying to keep some form of separation of concerns by making a separate render function for condensed but actually think that was a bit silly now. I'm going to make a mode flag, and like you suggested, use the mobile version.

varffs reviewed (changes requested):
- Remembered we need to do a pass over this code for accessibility issues
- A few queries about the function of the condensed version
- A text display bug (maybe just on no data migration state but worth catching)
- Grid system css detail

kimberleyworld committed:
Changed html() to text() in support.js

kimberleyworld committed:
Swopped to php comments

kimberleyworld committed:
Swopped to curly conditionals

kimberleyworld committed:
added a default and curley brackets

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in a899d86 according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

kimberleyworld committed:
Moved definitions to top

kimberleyworld committed:
Merge branch 'feature-update-donate-form' of https://github.com/novaramedia/novaramedia-com into feature-update-donate-form

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 8920ebe according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

kimberleyworld committed:
Added tab nav and ensured first button seleced

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in b0cb598 according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

kimberleyworld committed:
added arrow navigation

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in f33a212 according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

kimberleyworld committed:
Fixed focus states as you navigate with arrows

kimberleyworld committed:
Merge branch 'feature-update-donate-form' of https://github.com/novaramedia/novaramedia-com into feature-update-donate-form

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in d72bdad according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

kimberleyworld committed:
Mde £ sign black when input active

kimberleyworld committed:
Merge branch 'feature-update-donate-form' of https://github.com/novaramedia/novaramedia-com into feature-update-donate-form

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in e9bc42b according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

kimberleyworld committed:
Fixed input staying highlighted when schedule was changed

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in ad387b8 according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

kimberleyworld committed:
added grid-item--tight to left and right schedule column

kimberleyworld committed:
Merge branch 'feature-update-donate-form' of https://github.com/novaramedia/novaramedia-com into feature-update-donate-form

kimberleyworld committed:
Added grid row in

kimberleyworld committed:
removed support-button--active in support js

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 41e27d5 according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

kimberleyworld committed:
use unified markup for banner and condensed variants with mode tag

kimberleyworld committed:
Merge branch 'feature-update-donate-form' of https://github.com/novaramedia/novaramedia-com into feature-update-donate-form

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 892933a according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

kimberleyworld committed:
Fixed hover state - border colour

kimberleyworld committed:
Merge branch 'feature-update-donate-form' of https://github.com/novaramedia/novaramedia-com into feature-update-donate-form

kimberleyworld committed:
Removed condensed styles

kimberleyworld committed:
Revert "Removed condensed styles"

This reverts commit d5276deb6066157919c979d7890802a25a294c1d.

kimberleyworld committed:
Fixed responsive condensed layout using grid items.

kimberleyworld committed:
fixed fading on mobile

varffs commented on src/js/modules/Support.js:
```diff
@@ -68,44 +84,127 @@ export class Support {
-
+          // function to update the support section copy depending on the type of donation
+          function updateSupportSection(data, $form) {
+            const heading = $form.find('.support-form__dynamic-heading');
```
variables declaring a jquery object should have the prefix $ for ease of reading. aka `$heading`

varffs commented on src/js/modules/Support.js:
```diff
@@ -68,44 +84,127 @@ export class Support {
           const data = $button.data();
-
+          // function to update the support section copy depending on the type of donation
+          function updateSupportSection(data, $form) {
```
this function should be a general method on the class for this functionality otherwise it will be being redeclared multiple times.

varffs commented on src/js/modules/Support.js:
```diff
@@ -50,16 +50,32 @@ export class Support {
+        $firstValueBtn
+          .addClass('ui-button--active')
+          .attr('aria-checked', 'true')
+          .attr('tabindex', '0');
```
to understand: why do you set the tab index here programatically?

kimberleyworld replied:
It's to manage keyboard accessibility for the custom radio button group, it mimics native radio buttons. Only the currently selected (active) button is focusable with the Tab key (tabindex="0"). All other buttons are not focusable (tabindex="-1").

varffs commented on src/styl/layouts/support-form.styl:
```diff
@@ -4,27 +4,122 @@
+    background-color: var(--color-white-soft);
+    border-color: var(--color-white-soft);
+
+.support-form__input-wrapper
```
if you just need this as a helper and no other styles on this element this is a perfect thing to add to the stylus library. I've documented a flow for how to do that while building out features here: https://github.com/novaramedia/nm-stylus-library?tab=readme-ov-file#releases

kimberleyworld replied:
I've added it to the design library but it will defo need to be renamed. I'll leave you to decide the same for such a specific thing

varffs commented on src/styl/layouts/support-form.styl:
```diff
@@ -4,27 +4,122 @@
-.support-form__button, .support-form__custom-input, .support-form__submit
-  color: var(--color-red)
+.support-form__schedule-option-right
+  border-radius:  0px 4px 4px 0px;
```
in v0.11.0 of stylus library there is a css variable for this. probably worth updating the dependency in this feature and using that var here

kimberleyworld replied:
note to come back and do this once the library has been updated and version advanced here

varffs commented on src/styl/layouts/support-form.styl:
```diff
@@ -4,27 +4,122 @@
+  .support-form__buttons-desktop
+    display: none;
+
+  .ui-button:active
```
is this needed in the library?

kimberleyworld replied:
Yes still needed for mobile, to hide the desktop components

varffs reviewed (changes requested):
Might be a bit Friday afternoon brain but could we potentially do a walkthru of the javascript together so I can better understand it.

There is also a bug when there are no values for the conditional copy and you toggle different schedule. This happens on my local:

<img width="548" height="382" alt="Screenshot 2025-07-25 at 16 54 46" src="https://github.com/user-attachments/assets/2d6875b0-d534-46d5-8efb-bfeb118dc214" />

kimberleyworld commented:
I've edited the JS to fix the 'false' error. I also noticed that if the override text was not set but the default was, the copy would still change based on donation type (regular/one off). Now, if the override text isn't fully set, the default is used and remains unchanged when switching types. If neither is set, the hardcoded value from the renderer is used. 

I use the data-value attribute on each donation type button ("regular", "oneoff"). When a button is clicked, its data-value is read as data.value in the JS. This value is used to look up the correct copy (heading/text) in WP.supportSectionCopy[data.value]. So, clicking a different button changes data.value, which switches the copy shown in the form based on the selected donation type. Let me know if you still want me to run through it.

kimberleyworld committed:
Fixed logic 'false' issue and jQuery variable definition

Modified src/js/modules/Support.j

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in c44254f according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

kimberleyworld committed:
updateSupportSection function refactored into a method on the Support class.

kimberleyworld committed:
Merge branch 'feature-update-donate-form' of https://github.com/novaramedia/novaramedia-com into feature-update-donate-form

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in d1dee7c according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

varffs committed:
isNonEmptyString refactor

use lodash and make utility function

varffs committed:
new flow for stylus library updating

varffs committed:
fix css PR issues

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 0a12fef according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

varffs committed:
fix icon render inconsistency

varffs committed:
refactor markup

varffs committed:
Create copilot-instructions.md

varffs committed:
partial refactor input classes

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 36e9146 according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

varffs committed:
refactor custom input active styles

varffs committed:
Merge branch 'feature-update-donate-form' of https://github.com/novaramedia/novaramedia-com into feature-update-donate-form

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in 4bc484d according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

varffs committed:
fix resizing submit type bug

varffs committed:
rename variables to non abbreviations

varffs committed:
make numberWithCommas a utility

varffs committed:
Update copilot-instructions.md

varffs committed:
gray schedule toggle as default

with parameter to set white for on red usage

varffs committed:
Merge branch 'feature-update-donate-form' of https://github.com/novaramedia/novaramedia-com into feature-update-donate-form

deepsource-autofix[bot] committed:
style: format code with Prettier

This commit fixes the style issues introduced in ef07aa6 according to the output
from Prettier.

Details: https://github.com/novaramedia/novaramedia-com/pull/347

varffs committed:
style tweaks

and fix undertargeted style

varffs committed:
last design details from figma

varffs committed:
refactor stylus file for readability

varffs committed:
refactor button styles

to minimise library one use classes

varffs committed:
update dep for security alert

varffs committed:
Update CHANGELOG.md

varffs committed:
fix layout and implementation

varffs committed:
refactor render_support_form_dispatcher

varffs committed:
refactor render to allow custom container classes

varffs committed:
refactor section variant BEM and render

varffs committed:
dont assume (NTS) the required container classes

varffs committed:
Merge branch 'development' into feature-update-donate-form
